treeherder/log_parser/crossreference.py

```diff
@@ -63,23 +62,23 @@ def _crossreference(job):
     summary = TextLogSummary.objects.create(job_guid=job.guid,
                                             repository=job.repository)
 
-    match_iter = structured_iterator(failure_lines)
-    failure_line, regexp = match_iter.next()
+    match_iter = structured_iterator(list(failure_lines.all()))
+    failure_line, _, fn = match_iter.next()
 
     summary_lines = []
 
     # For each error in the text log, try to match the next unmatched
     # structured log line
-    for error in text_log_errors:
-        if regexp and regexp.match(error.line.strip()):
+    for error in list(text_log_errors.all()):
```

Comment:
That was intentional because fetching the lines one at a time seemed needlessly inefficient from a db point of view. Happy to revert this if you think it's the wrong tradeoff.